Userland/Applications/Run/RunWindow.cpp

```diff
@@ -120,6 +114,11 @@ bool RunWindow::run_as_command(ByteString const& run_input)
 
     pid_t child_pid = maybe_child_pid.release_value();
 
+    // The program was able to start. Let's add to the history list immediately so users can see it as the first entry.
+    prepend_history(run_input);
+    // FIXME: Handle failure to save history somehow.
+    (void)save_history();
```

Comment:
Should we just propagate errors to serenity_main() or just continue to ignore failures to load/save history?

Reply:
well, I'm not sure what we could do with the error. Since it's not normally going to be launched from a terminal, It will probably end up getting swallowed either way. Maybe once we have a nice logging API we could log the failure to a syslog-like interface.